server.h

```diff
@@ -955,17 +978,23 @@ class CTFOServer final {
 
               comments_mutator.Add(comment);
 
-              // Emit the "new comment" notification.
+              // Emit the "new comment on my card" notification.
               if (card_author_uid != UID::INVALID_USER && card_author_uid != uid) {
-                data.notifications.Add(Notification(
-                    card_author_uid, now, std::make_shared<NotificationMyCardNewComment>(comment)));
+                // Do not send "new comment on my card" if it is the reply to my own comment.
+                if (parent_comment_author_uid == UID::INVALID_USER ||
+                    parent_comment_author_uid != card_author_uid) {
```

Comment:
Yes. Just like clarity sometimes. What if whoever reads this code is a Java guy who forgot how to use their brains years ago?